plugin/trino-hive/src/main/java/io/trino/plugin/hive/GenericHiveRecordCursorProvider.java

```diff
@@ -100,13 +100,22 @@ public GenericHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, DataSize
                     schema,
                     readerColumns);
 
-            return new GenericHiveRecordCursor<>(
-                    configuration,
-                    path,
-                    genericRecordReader(recordReader),
-                    length,
-                    schema,
-                    readerColumns);
+            try {
+                return new GenericHiveRecordCursor<>(
+                        configuration,
+                        path,
+                        genericRecordReader(recordReader),
+                        length,
+                        schema,
+                        readerColumns);
+            } catch (Exception e) {
+                try {
+                    recordReader.close();
+                } catch (IOException closeException) {
+                    e.addSuppressed(closeException);
```

Comment:
add `if e != closeException`, as the underlying code may throw same exception twice (not a good idea, i know, but apparently may happen)

Reply:
For some reason intellij claims that `(e != closeException)` is always true. Weird.

Reply:
It's really unfortunate that they didn't simply ignore self-suppression. Maybe they could fix that, or add a new `addSuppressedIgnore()` method...

Reply:
In a code like this it's not a big deal to add an `if` (but one still can think about this, which is totally sad). Somewhat bigger problem is when you have try-with-resources and `close()` rethrows the same exception as the one thrown by `try` (as in https://github.com/trinodb/trino/issues/3326), because there is nowhere to add an `if` to fix that...